GitHubContents.ps1

```diff
@@ -143,3 +153,266 @@ function Get-GitHubContent
+            Mandatory,
+            ParameterSetName = 'Elements')]
+        [string] $RepositoryName,
+
+        [string] $BranchName,
+
+        [string] $CommitterName,
+
+        [string] $CommitterEmail,
+
+        [string] $AuthorName,
+
+        [string] $AuthorEmail,
+
+        [string] $AccessToken,
+
+        [switch] $NoStatus
+    )
+
+    $elements = Resolve-RepositoryElements -DisableValidation
+    $OwnerName = $elements.ownerName
+    $RepositoryName = $elements.repositoryName
+
+    $telemetryProperties = @{
+        'OwnerName' = (Get-PiiSafeString -PlainText $OwnerName)
+        'RepositoryName' = (Get-PiiSafeString -PlainText $RepositoryName)
+    }
+
+    $uriFragment = "/repos/$OwnerName/$RepositoryName/contents/$Path"
+
+    $encodedContent = [Convert]::ToBase64String([System.Text.Encoding]::Unicode.GetBytes($Content))
+
+    $hashBody = @{
+        message = $CommitMessage
+        content = $encodedContent
+    }
+
+    if ($PSBoundParameters.ContainsKey('BranchName'))
+    {
+        $hashBody['branch'] = $BranchName
+    }
+
+    if ($PSBoundParameters.ContainsKey('CommitterName') -or
+        $PSBoundParameters.ContainsKey('CommitterEmail'))
+    {
+        if (![System.String]::IsNullOrEmpty($CommitterName) -and
+            ![System.String]::IsNullOrEmpty($CommitterEmail))
+        {
+            $hashBody['committer'] = @{
+                name = $CommitterName
+                email = $CommitterEmail
+            }
+        }
+        else
+        {
+            throw 'Both CommiterName and CommitterEmail need to be specified.'
+        }
+    }
+
+    if ($PSBoundParameters.ContainsKey('AuthorName') -or
+        $PSBoundParameters.ContainsKey('AuthorEmail'))
+    {
+        if (![System.String]::IsNullOrEmpty($CommitterName) -and
+            ![System.String]::IsNullOrEmpty($CommitterEmail))
+        {
+            $hashBody['author'] = @{
+                name = $AuthorName
+                email = $AuthorEmail
+            }
+        }
+        else
+        {
+            throw 'Both AuthorName and AuthorEmail need to be specified.'
+        }
+    }
+
+    if ($PSCmdlet.ShouldProcess("$BranchName branch of $RepositoryName",
+        "Set GitHub Contents on $Path"))
+    {
+        Write-InvocationLog
+
+        $params = @{
+            UriFragment = $uriFragment
+            Description = "Writing content for $Path in the $BranchName branch of $RepositoryName"
+            Body = (ConvertTo-Json -InputObject $hashBody)
+            Method = 'Put'
+            AccessToken = $AccessToken
+            TelemetryEventName = $MyInvocation.MyCommand.Name
+            TelemetryProperties = $telemetryProperties
+            NoStatus = (Resolve-ParameterWithDefaultConfigurationValue -Name NoStatus `
+                -ConfigValueName DefaultNoStatus)
+        }
+
+        try
+        {
+            return Invoke-GHRestMethod @params
+        }
+        catch
+        {
+            $overwriteShaRequired = $false
```

Comment:
This is all well and good, but I think you should still explicitly expose a `Sha` parameter.  That way this extra check can be avoided if a user pipes in a `GitHub.Content` object which already has the sha and all of the other info).

Reply:
Done